<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>1.1 Introduction to Coding in R | The inTelligence And Machine lEarning (TAME) Toolkit for Introductory Data Science, Chemical-Biological Analyses, Predictive Modeling, and Database Mining for Environmental Health Research</title>
  <meta name="description" content="1.1 Introduction to Coding in R | The inTelligence And Machine lEarning (TAME) Toolkit for Introductory Data Science, Chemical-Biological Analyses, Predictive Modeling, and Database Mining for Environmental Health Research" />
  <meta name="generator" content="bookdown 0.25 and GitBook 2.6.7" />

  <meta property="og:title" content="1.1 Introduction to Coding in R | The inTelligence And Machine lEarning (TAME) Toolkit for Introductory Data Science, Chemical-Biological Analyses, Predictive Modeling, and Database Mining for Environmental Health Research" />
  <meta property="og:type" content="book" />
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="1.1 Introduction to Coding in R | The inTelligence And Machine lEarning (TAME) Toolkit for Introductory Data Science, Chemical-Biological Analyses, Predictive Modeling, and Database Mining for Environmental Health Research" />
  
  
  

<meta name="author" content="Kyle R. Roell, Lauren E. Koval, Rebecca Boyles, Grace Patlewicz, Caroline Ring, Cynthia Rider, Cavin Ward-Caviness, David M. Reif, Ilona Jaspers, Rebecca C. Fry, and Julia E. Rager" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  <link rel="shortcut icon" href="images/Favicon_v1.png" type="image/x-icon" />
<link rel="prev" href="index.html"/>
<link rel="next" href="data-organization-basics.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="toc-logo"><a href="./"><img src="images/Data Training Icon for Top Left Bookdown_v2.png"><br></a></li>

<li class="divider"></li>
<li><a href="index.html#preface">Preface<span></span></a></li>
<li class="part"><span><b>Chapter 1 Introductory <br>Data Science<span></span></b></span></li>
<li><a href="introduction-to-coding-in-r.html#introduction-to-coding-in-r">1.1 Introduction to Coding in R<span></span></a>
<ul>
<li><a href="introduction-to-coding-in-r.html#introduction-to-training-module">Introduction to Training Module<span></span></a></li>
<li><a href="introduction-to-coding-in-r.html#r-packages">R Packages<span></span></a></li>
<li><a href="introduction-to-coding-in-r.html#scripting-basics">Scripting Basics<span></span></a></li>
<li><a href="introduction-to-coding-in-r.html#concluding-remarks-1">Concluding Remarks<span></span></a></li>
</ul></li>
<li><a href="data-organization-basics.html#data-organization-basics">1.2 Data Organization Basics<span></span></a>
<ul>
<li><a href="data-organization-basics.html#introduction-to-training-module-1">Introduction to Training Module<span></span></a></li>
<li><a href="data-organization-basics.html#data-manipulation-using-base-r">Data Manipulation using Base R<span></span></a></li>
<li><a href="data-organization-basics.html#introduction-to-tidyverse">Introduction to Tidyverse<span></span></a></li>
<li><a href="data-organization-basics.html#concluding-remarks-2">Concluding Remarks<span></span></a></li>
</ul></li>
<li><a href="finding-and-visualizing-data-trends.html#finding-and-visualizing-data-trends">1.3 Finding and Visualizing Data Trends<span></span></a>
<ul>
<li><a href="finding-and-visualizing-data-trends.html#introduction-to-training-module-2">Introduction to Training Module<span></span></a></li>
<li><a href="finding-and-visualizing-data-trends.html#basic-data-analysis">Basic Data Analysis<span></span></a></li>
<li><a href="finding-and-visualizing-data-trends.html#regression-modeling">Regression Modeling<span></span></a></li>
<li><a href="finding-and-visualizing-data-trends.html#categorical-data-analysis">Categorical Data Analysis<span></span></a></li>
<li><a href="finding-and-visualizing-data-trends.html#concluding-remarks-3">Concluding Remarks<span></span></a></li>
</ul></li>
<li><a href="high-dimensional-data-visualizations.html#high-dimensional-data-visualizations">1.4 High-Dimensional Data Visualizations<span></span></a>
<ul>
<li><a href="high-dimensional-data-visualizations.html#introduction-to-training-module-3">Introduction to Training Module<span></span></a></li>
<li><a href="high-dimensional-data-visualizations.html#high-dimensional-data-visualizations-1">High-Dimensional Data Visualizations<span></span></a></li>
<li><a href="high-dimensional-data-visualizations.html#density-plot-visualizations">Density Plot Visualizations<span></span></a></li>
<li><a href="high-dimensional-data-visualizations.html#ggally-visualizations">GGally Visualizations<span></span></a></li>
<li><a href="high-dimensional-data-visualizations.html#boxplot-visualizations">Boxplot Visualizations<span></span></a></li>
<li><a href="high-dimensional-data-visualizations.html#correlation-plot-visualizations">Correlation Plot Visualizations<span></span></a></li>
<li><a href="high-dimensional-data-visualizations.html#hierarchical-clustering-visualizations">Hierarchical Clustering Visualizations<span></span></a>
<ul>
<li><a href="high-dimensional-data-visualizations.html#heat-map-visualizations">Heat Map Visualizations<span></span></a></li>
</ul></li>
<li><a href="high-dimensional-data-visualizations.html#concluding-remarks-4">Concluding Remarks<span></span></a></li>
</ul></li>
<li><a href="fair-data-management-practices.html#fair-data-management-practices">1.5 FAIR Data Management Practices<span></span></a>
<ul>
<li><a href="fair-data-management-practices.html#introduction-to-training-module-4">Introduction to Training Module<span></span></a></li>
<li><a href="fair-data-management-practices.html#introduction-to-fair">Introduction to FAIR<span></span></a></li>
<li><a href="fair-data-management-practices.html#breaking-down-fair-letter-by-letter">Breaking Down FAIR, Letter-by-Letter<span></span></a></li>
<li><a href="fair-data-management-practices.html#data-repositories-for-sharing-of-data">Data Repositories for Sharing of Data<span></span></a></li>
<li><a href="fair-data-management-practices.html#helpful-resources-on-fair">Helpful Resources on FAIR<span></span></a></li>
<li><a href="fair-data-management-practices.html#concluding-remarks-5">Concluding Remarks<span></span></a></li>
</ul></li>
<li class="part"><span><b>Chapter 2 Chemical-<br>Biological Analyses <br>and Predictive Modeling<span></span></b></span></li>
<li><a href="dose-response-modeling.html#dose-response-modeling">2.1 Dose-Response Modeling<span></span></a>
<ul>
<li><a href="dose-response-modeling.html#introduction-to-training-module-5">Introduction to Training Module<span></span></a></li>
<li><a href="dose-response-modeling.html#plotting-data-in-dose-response">Plotting Data in Dose-Response<span></span></a></li>
<li><a href="dose-response-modeling.html#dose-response-curve-fitting">Dose-Response Curve Fitting<span></span></a></li>
<li><a href="dose-response-modeling.html#comparing-curve-fits">Comparing Curve Fits<span></span></a></li>
<li><a href="dose-response-modeling.html#curve-fitting-example">Curve Fitting Example<span></span></a></li>
<li><a href="dose-response-modeling.html#deriving-benchmark-dose">Deriving Benchmark Dose<span></span></a></li>
<li><a href="dose-response-modeling.html#concluding-remarks-6">Concluding Remarks<span></span></a></li>
</ul></li>
<li><a href="machine-learning-and-predictive-modeling.html#machine-learning-and-predictive-modeling">2.2 Machine Learning and Predictive Modeling<span></span></a>
<ul>
<li><a href="machine-learning-and-predictive-modeling.html#introduction-to-training-module-6">Introduction to Training Module<span></span></a></li>
<li><a href="machine-learning-and-predictive-modeling.html#k-means-analysis">K-means Analysis<span></span></a></li>
<li><a href="machine-learning-and-predictive-modeling.html#principal-component-analysis">Principal Component Analysis<span></span></a></li>
<li><a href="machine-learning-and-predictive-modeling.html#combining-k-means-with-pca">Combining K-Means with PCA<span></span></a></li>
<li><a href="machine-learning-and-predictive-modeling.html#concluding-remarks-7">Concluding Remarks<span></span></a></li>
</ul></li>
<li><a href="mixtures-analyses.html#mixtures-analyses">2.3 Mixtures Analyses<span></span></a>
<ul>
<li><a href="mixtures-analyses.html#introduction-to-training-module-7">Introduction to Training Module<span></span></a></li>
<li><a href="mixtures-analyses.html#chemistry-based-approach">Chemistry-based Approach<span></span></a></li>
<li><a href="mixtures-analyses.html#toxicity-based-approach">Toxicity-based Approach<span></span></a></li>
<li><a href="mixtures-analyses.html#comparing-results">Comparing Results<span></span></a></li>
<li><a href="mixtures-analyses.html#concluding-remarks-8">Concluding Remarks<span></span></a></li>
</ul></li>
<li><a href="omics-analyses-and-systems-biology.html#omics-analyses-and-systems-biology">2.4 -Omics Analyses and Systems Biology<span></span></a>
<ul>
<li><a href="omics-analyses-and-systems-biology.html#introduction-to-training-module-8">Introduction to Training Module<span></span></a></li>
<li><a href="omics-analyses-and-systems-biology.html#transcriptomics-data-qaqc">Transcriptomics Data QA/QC<span></span></a></li>
<li><a href="omics-analyses-and-systems-biology.html#statistical-analysis-of-gene-expression">Statistical Analysis of Gene Expression<span></span></a></li>
<li><a href="omics-analyses-and-systems-biology.html#ma-plots">MA Plots<span></span></a></li>
<li><a href="omics-analyses-and-systems-biology.html#volcano-plots">Volcano Plots<span></span></a></li>
<li><a href="omics-analyses-and-systems-biology.html#pathway-enrichment-analysis">Pathway Enrichment Analysis<span></span></a></li>
<li><a href="omics-analyses-and-systems-biology.html#concluding-remarks-9">Concluding Remarks<span></span></a></li>
</ul></li>
<li><a href="toxicokinetic-modeling.html#toxicokinetic-modeling">2.5 Toxicokinetic Modeling<span></span></a>
<ul>
<li><a href="toxicokinetic-modeling.html#introduction-to-training-module-9">Introduction to Training Module<span></span></a></li>
<li><a href="toxicokinetic-modeling.html#data-and-models-used-in-toxicokinetic-modeling-tk">Data and Models used in Toxicokinetic Modeling (TK)<span></span></a></li>
<li><a href="toxicokinetic-modeling.html#calculating-steady-state-concentration">Calculating steady-state concentration<span></span></a></li>
<li><a href="toxicokinetic-modeling.html#reverse-toxicokinetics">Reverse Toxicokinetics<span></span></a></li>
<li><a href="toxicokinetic-modeling.html#monte-carlo-approach">Monte Carlo Approach<span></span></a></li>
<li><a href="toxicokinetic-modeling.html#chemical-specific-example">Chemical-Specific Example<span></span></a></li>
<li><a href="toxicokinetic-modeling.html#calculating-bioactivity-exposure-ratios-bers">Calculating Bioactivity-Exposure Ratios (BERs)<span></span></a></li>
<li><a href="toxicokinetic-modeling.html#concluding-remarks-10">Concluding Remarks<span></span></a></li>
</ul></li>
<li><a href="read-across-toxicity-predictions.html#read-across-toxicity-predictions">2.6 Read-Across Toxicity Predictions<span></span></a>
<ul>
<li><a href="read-across-toxicity-predictions.html#introduction-to-training-module-10">Introduction to Training Module<span></span></a></li>
<li><a href="read-across-toxicity-predictions.html#read-across-example-analysis">Read-Across Example Analysis<span></span></a></li>
<li><a href="read-across-toxicity-predictions.html#calculating-chemical-similarities">Calculating Chemical Similarities<span></span></a></li>
<li><a href="read-across-toxicity-predictions.html#chemical-read-across-to-predict-acute-toxicity">Chemical Read-Across to Predict Acute Toxicity<span></span></a></li>
<li><a href="read-across-toxicity-predictions.html#concluding-remarks-11">Concluding Remarks<span></span></a></li>
</ul></li>
<li class="part"><span><b>Chapter 3 Environmental <br>Health Database Mining<span></span></b></span></li>
<li><a href="comparative-toxicogenomics-database.html#comparative-toxicogenomics-database">3.1 Comparative Toxicogenomics Database<span></span></a>
<ul>
<li><a href="comparative-toxicogenomics-database.html#introduction-to-training-module-11">Introduction to Training Module<span></span></a></li>
<li><a href="comparative-toxicogenomics-database.html#ctd-data-in-r">CTD Data in R<span></span></a></li>
<li><a href="comparative-toxicogenomics-database.html#identifying-genes-under-epigenetic-control">Identifying Genes under Epigenetic Control<span></span></a></li>
<li><a href="comparative-toxicogenomics-database.html#concluding-remarks-12">Concluding Remarks<span></span></a></li>
</ul></li>
<li><a href="gene-expression-omnibus.html#gene-expression-omnibus">3.2 Gene Expression Omnibus<span></span></a>
<ul>
<li><a href="gene-expression-omnibus.html#introduction-to-training-module-12">Introduction to Training Module<span></span></a>
<ul>
<li><a href="gene-expression-omnibus.html#training-modules-environmental-health-questions-7">Training Module’s <strong>Environmental Health Questions</strong><span></span></a></li>
</ul></li>
<li><a href="gene-expression-omnibus.html#geo-data-in-r">GEO Data in R<span></span></a></li>
<li><a href="gene-expression-omnibus.html#visualizing-data">Visualizing Data<span></span></a></li>
<li><a href="gene-expression-omnibus.html#statistical-analyses">Statistical Analyses<span></span></a></li>
<li><a href="gene-expression-omnibus.html#concluding-remarks-13">Concluding Remarks<span></span></a></li>
</ul></li>
<li><a href="database-integration-air-quality-mortality-and-environmental-justice-data.html#database-integration-air-quality-mortality-and-environmental-justice-data">3.3 Database Integration: Air Quality, Mortality, and Environmental Justice Data<span></span></a>
<ul>
<li><a href="database-integration-air-quality-mortality-and-environmental-justice-data.html#introduction-to-training-module-13">Introduction to Training Module<span></span></a></li>
<li><a href="database-integration-air-quality-mortality-and-environmental-justice-data.html#population-weighted-vs-unweighted-exposures">Population-weighted vs Unweighted Exposures<span></span></a></li>
<li><a href="database-integration-air-quality-mortality-and-environmental-justice-data.html#regression-modeling-1">Regression Modeling<span></span></a></li>
<li><a href="database-integration-air-quality-mortality-and-environmental-justice-data.html#environmental-justice-considerations">Environmental Justice Considerations<span></span></a></li>
<li><a href="database-integration-air-quality-mortality-and-environmental-justice-data.html#concluding-remarks-14">Concluding Remarks<span></span></a></li>
</ul></li>
<li class="part"><span><b>Additional Resources<span></span></b></span></li>
<li><a href="resources.html#resources">Resources<span></span></a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">The inTelligence And Machine lEarning (TAME) Toolkit for Introductory Data Science, Chemical-Biological Analyses, Predictive Modeling, and Database Mining for Environmental Health Research</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="introduction-to-coding-in-r" class="section level1 hasAnchor">
<h1>1.1 Introduction to Coding in R<a href="introduction-to-coding-in-r.html#introduction-to-coding-in-r" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>This training module was developed by Dr. Kyle R. Roell and Dr. Julia E. Rager</p>
<p>Fall 2021</p>
<div id="introduction-to-training-module" class="section level2 hasAnchor">
<h2>Introduction to Training Module<a href="introduction-to-coding-in-r.html#introduction-to-training-module" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="what-is-r" class="section level4 hasAnchor">
<h4>What is R?<a href="introduction-to-coding-in-r.html#what-is-r" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Computer script can be used to increase data analysis reproducibility, transparency, and methods sharing, and is becoming increasingly incorporated into exposure science, toxicology, and environmental health research. One of the most utilized coding language in this research field is the <strong>R coding language</strong>. Some advantages of using R include the following:</p>
<ul>
<li>Free, open-source programming language that is licensed under the Free Software Foundation’s GNU General Public License</li>
<li>Can be run across all major platforms and operating systems, including Unix, Windows, and MacOS</li>
<li>Publicly available packages help you carry out analyses efficiently (without having to code for everything yourself)</li>
<li>Large, diverse collection of packages</li>
<li>Comprehensive documentation</li>
<li>When code is efficiently tracked during development/execution, it promotes reproducible analyses</li>
</ul>
<p>Because of these advantages, R has emerged as an avenue for world-wide collaboration in data science.</p>
</div>
<div id="downloading-and-installing-r" class="section level4 hasAnchor">
<h4>Downloading and Installing R<a href="introduction-to-coding-in-r.html#downloading-and-installing-r" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>First, users should download R by navigating to the following website: <a href="https://www.r-project.org/" class="uri">https://www.r-project.org/</a></p>
<p>And then clicking the ‘download R’ link:
<img src="_book/TAME_Toolkit_files/figure-html/Module1_1_DownloadR_Image1.png" width="696" /></p>
<p>This link will navigate you to the CRAN mirror website. Click on the CRAN mirror location that seems closest to your usual location:
<img src="_book/TAME_Toolkit_files/figure-html/Module1_1_DownloadR_Image2.png" width="696" /></p>
<p>This will lead you to the selected CRAN Network website (here, <a href="https://archive.linux.duke.edu/cran/" class="uri">https://archive.linux.duke.edu/cran/</a>), where you will again select a download option:</p>
<p><img src="_book/TAME_Toolkit_files/figure-html/Module1_1_DownloadR_Image3.png" width="696" /></p>
<p>Then, select the top (representing the most recent) available .pkg file to download, and then install according to your computer’s typical program installation steps.</p>
<p><img src="_book/TAME_Toolkit_files/figure-html/Module1_1_DownloadR_Image4.png" width="696" /></p>
</div>
<div id="downloading-and-installing-r-studio" class="section level4 hasAnchor">
<h4>Downloading and Installing R Studio<a href="introduction-to-coding-in-r.html#downloading-and-installing-r-studio" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p><strong>What is R Studio?</strong>
RStudio is an Integrated Development Environment (IDE) for R, which makes it more ‘user friendly’ when developing and using R script.</p>
<p>RStudio Desktop is a desktop application that can be downloaded for free, online. To download RStudio:</p>
<ul>
<li>Navigate to: <a href="https://rstudio.com/products/rstudio/download/" class="uri">https://rstudio.com/products/rstudio/download/</a></li>
<li>Select the free RStudio Desktop option, and click “DOWNLOAD”</li>
<li>Then download the top (most recent) RStudio Desktop option for your operating system</li>
</ul>
<p>Here is a screenshot of what R script looks like within RStudio:</p>
<p><img src="_book/TAME_Toolkit_files/figure-html/Module1_1_RStudio_Image.png" width="684" /></p>
<!-- ## Introduction to R Packages -->
</div>
</div>
<div id="r-packages" class="section level2 hasAnchor">
<h2>R Packages<a href="introduction-to-coding-in-r.html#r-packages" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>One of the major benefits to coding in the R language is access to the continually expanding resource of thousands of user-developed packages that aid in improved data analyses and methods sharing. Packages have varying utilities, spanning basic organization and manipulation of data, visualizing data, and more advanced approaches to parse and analyze data, with examples included in all of the proceeding training modules.</p>
<p>In brief, packages represent compilations of code fitted for a specialized focus or purpose. These are
often written by R users and submitted to the <a href="https://cran.r-project.org/web/packages/">CRAN</a>, or another host such as <a href="https://www.bioconductor.org/">BioConductor</a> or <a href="https://github.com/">Github</a>.</p>
<p>Examples of some common packages that we’ll be using throughout these training modules include the following:</p>
<ul>
<li>tidyverse: A collection of open source R packages that share an underlying design philosophy, grammar, and data structures of tidy data. For more information on the tidyverse package, see its associated <a href="https://cran.r-project.org/web/packages/tidyverse/index.html">CRAN webpage</a>, primary <a href="https://www.tidyverse.org/packages/">webpage</a>, and peer-reviewed <a href="https://onlinelibrary.wiley.com/doi/10.1002/sdr.1600">article released in 2018</a>.</li>
<li>ggplot2: A system, within the tidyverse collection, for declaratively creating graphics. Users provide data they would like to plot, and then control how ggplot2 should map variables to aesthetics and define which graphical primitives to use; ggplot2 then drafts the corresponding visualizations. For more information on the ggplot2 package, see its associated <a href="https://cran.r-project.org/web/packages/ggplot2/index.html">CRAN webpage</a> and <a href="https://www.rdocumentation.org/packages/ggplot2/versions/3.3.5">R Documentation</a>.</li>
</ul>
<p>More information on these packages, as well as many others, is included throughout the training modules.</p>
<div id="downloadinginstalling-r-packages" class="section level4 hasAnchor">
<h4>Downloading/Installing R Packages<a href="introduction-to-coding-in-r.html#downloadinginstalling-r-packages" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>R packages often do not need to be downloaded from a website. Instead, you can just load packages through running script in R, like:</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="introduction-to-coding-in-r.html#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">install.packages</span>(“tidyverse”)</span>
<span id="cb1-2"><a href="introduction-to-coding-in-r.html#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span></code></pre></div>
<p>It is worth noting that a function can be queried in RStudio by typing a question mark before the name of the function. For example:</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="introduction-to-coding-in-r.html#cb2-1" aria-hidden="true" tabindex="-1"></a>?install.packages</span></code></pre></div>
<p>This will bring up documentation in the viewer window.</p>
</div>
</div>
<div id="scripting-basics" class="section level2 hasAnchor">
<h2>Scripting Basics<a href="introduction-to-coding-in-r.html#scripting-basics" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="comments" class="section level4 hasAnchor">
<h4>Comments<a href="introduction-to-coding-in-r.html#comments" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>R allows for scripts to contain non-code elements, called comments, that will not be run or interpreted. Comments are commonly included when annotating code, or describing what your code does, where your data came from, and just general textual reminders throughout the script.</p>
<p>To make a comment, simply use a # followed by the comment.
A # only comments out a single line of code. In other words, only that line will be commented and therefore not be run, but lines directly above/below it still will be</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="introduction-to-coding-in-r.html#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="co"># This is an R comment!</span></span></code></pre></div>
<p>Comments are useful to help make code more interpretable for others or to add reminders of what and why parts of code may have been written.</p>
</div>
<div id="starting-code" class="section level4 hasAnchor">
<h4>Starting Code<a href="introduction-to-coding-in-r.html#starting-code" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>RStudio will autofill function names, variable names, etc. by pressing tab while typing. If multiple matches are found, RStudio will provide you with a drop down list to select from, which may be useful when searching through newly installed packages or trying to quickly type variable names in an R script.</p>
<p>One of the first lines of code in any script will likely include the loading of packages needed to run the script. Here is an example line of code to load a package:</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="introduction-to-coding-in-r.html#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Loading ggplot2 package (should already be installed as a base package)</span></span>
<span id="cb4-2"><a href="introduction-to-coding-in-r.html#cb4-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span></code></pre></div>
<p>Many packages also exist as part of the baseline configuration of an R working environment, and do not require manual loading each time you launch R. These include the following packages:</p>
<ul>
<li>datasets</li>
<li>graphics</li>
<li>methods</li>
<li>stats</li>
<li>utils</li>
</ul>
</div>
<div id="setting-your-working-directory" class="section level4 hasAnchor">
<h4>Setting Your Working Directory<a href="introduction-to-coding-in-r.html#setting-your-working-directory" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Another step that is commonly done at the very beginning of your code is setting your working direction. This points to where you have files that you want to upload / where the default is to deposit output files produced during your scripted activities.</p>
<p>You must set the working directory to a local directory where data are located or output files will be saved.</p>
<p>To view where your current working directory is (by default), run the following:</p>
</div>
<div id="show-your-working-directory" class="section level4 hasAnchor">
<h4>Show your working directory<a href="introduction-to-coding-in-r.html#show-your-working-directory" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="introduction-to-coding-in-r.html#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="fu">getwd</span>()</span></code></pre></div>
<p>To set the location of your working directory, run the following:</p>
</div>
<div id="set-your-working-directory" class="section level4 hasAnchor">
<h4>Set your working directory<a href="introduction-to-coding-in-r.html#set-your-working-directory" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="introduction-to-coding-in-r.html#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="fu">setwd</span>(<span class="st">&quot;/filepath to where your input files are&quot;</span>)</span></code></pre></div>
<p>Note that in macOS, filepaths use “/” as folder separaters; whereas in PCs, filepaths use “\”.</p>
</div>
<div id="importing-files" class="section level4 hasAnchor">
<h4>Importing Files<a href="introduction-to-coding-in-r.html#importing-files" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>After setting the working directory, importing and exporting files can be done using various functions based on the type of file being read or written. Often, it is easiest to import data into R that are in a comma separated values / comma delimited file (.csv) or tab / text delimited file (.txt).</p>
<p>Other data types such as SAS data files, large .csv files, etc. may require different functions to be more efficiently read in and some of these file formats will be discussed in future modules.</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="introduction-to-coding-in-r.html#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Read in the .csv data that&#39;s located in our working directory</span></span>
<span id="cb7-2"><a href="introduction-to-coding-in-r.html#cb7-2" aria-hidden="true" tabindex="-1"></a>csv.dataset <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">&quot;Module1_1/Module1_1_ExampleData.csv&quot;</span>)</span>
<span id="cb7-3"><a href="introduction-to-coding-in-r.html#cb7-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-4"><a href="introduction-to-coding-in-r.html#cb7-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Read in the .txt data</span></span>
<span id="cb7-5"><a href="introduction-to-coding-in-r.html#cb7-5" aria-hidden="true" tabindex="-1"></a>txt.dataset <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">&quot;Module1_1/Module1_1_ExampleData.txt&quot;</span>)</span></code></pre></div>
<p>These datasets now appear as saved dataframes (“csv.dataset” and “txt.dataset”) in our working environment in R.</p>
</div>
<div id="viewing-data" class="section level4 hasAnchor">
<h4>Viewing Data<a href="introduction-to-coding-in-r.html#viewing-data" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>After data have been loaded into R, or created within R, you will likely want to view what these datasets look like.
Datasets can be viewed in their entirety, or datasets can be subsetted to quickly look at part of the data.</p>
<p>Here’s some example script to view just the beginnings of a dataframe, using the “head” function in R (a part of the baseline packages)</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="introduction-to-coding-in-r.html#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(csv.dataset)</span></code></pre></div>
<pre><code>##    Sample Var1 Var2 Var3
## 1 sample1    1    2    1
## 2 sample2    2    4    4
## 3 sample3    3    6    9
## 4 sample4    4    8   16
## 5 sample5    5   10   25</code></pre>
<p>Here, you can see that this automatically brings up a view of the first five rows of the dataframe.</p>
<p>Another way to view the first five rows of a dataframe is to run the following:</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="introduction-to-coding-in-r.html#cb10-1" aria-hidden="true" tabindex="-1"></a>csv.dataset[<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>,]</span></code></pre></div>
<pre><code>##    Sample Var1 Var2 Var3
## 1 sample1    1    2    1
## 2 sample2    2    4    4
## 3 sample3    3    6    9
## 4 sample4    4    8   16
## 5 sample5    5   10   25</code></pre>
<p>Expanding on this, to view the first 5 rows x 2 columns, run the following:</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="introduction-to-coding-in-r.html#cb12-1" aria-hidden="true" tabindex="-1"></a>csv.dataset[<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>,<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>]</span></code></pre></div>
<pre><code>##    Sample Var1
## 1 sample1    1
## 2 sample2    2
## 3 sample3    3
## 4 sample4    4
## 5 sample5    5</code></pre>
<p>To view the entire dataset in RStudio, use the “View” function:</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="introduction-to-coding-in-r.html#cb14-1" aria-hidden="true" tabindex="-1"></a><span class="fu">View</span>(csv.dataset)</span></code></pre></div>
</div>
<div id="exporting-data" class="section level4 hasAnchor">
<h4>Exporting Data<a href="introduction-to-coding-in-r.html#exporting-data" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Now that we have these datasets saved as dataframes, we can use these as examples to export data files from the R environment back into our local directory.</p>
<p>There are many ways to export data in R. Data can be written out into a .csv file, tab delimited .txt file, RData file, etc. There are also many functions within packages that write out specific datasets generated by that package.
To write out to a .csv file:</p>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="introduction-to-coding-in-r.html#cb15-1" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(csv.dataset, <span class="st">&quot;Module1_1_SameCSVFileNowOut.csv&quot;</span>)</span></code></pre></div>
<p>To write out a .txt tab delimited file:</p>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="introduction-to-coding-in-r.html#cb16-1" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(txt.dataset, <span class="st">&quot;Module1_1_SameTXTFileNowOut.txt&quot;</span>)</span></code></pre></div>
<p>R also allows objects to be saved in RData files. These files can be read into R, as well, and will load the object into the current workspace. Entire workspaces are also able to be saved.</p>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb17-1"><a href="introduction-to-coding-in-r.html#cb17-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Read in saved single R data object (note that this file is not provided, just example code is for future reference)</span></span>
<span id="cb17-2"><a href="introduction-to-coding-in-r.html#cb17-2" aria-hidden="true" tabindex="-1"></a>r.obj <span class="ot">=</span> <span class="fu">readRDS</span>(<span class="st">&quot;data.rds&quot;</span>)</span>
<span id="cb17-3"><a href="introduction-to-coding-in-r.html#cb17-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-4"><a href="introduction-to-coding-in-r.html#cb17-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Write single R object to file (note that this file is not provided, just example code is for future reference)</span></span>
<span id="cb17-5"><a href="introduction-to-coding-in-r.html#cb17-5" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(object, <span class="st">&quot;single_object.rds&quot;</span>)</span>
<span id="cb17-6"><a href="introduction-to-coding-in-r.html#cb17-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-7"><a href="introduction-to-coding-in-r.html#cb17-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Read in multiple saved R objects (note that this file is not provided, just example code is for future reference)</span></span>
<span id="cb17-8"><a href="introduction-to-coding-in-r.html#cb17-8" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="st">&quot;multiple_data.RData&quot;</span>)</span>
<span id="cb17-9"><a href="introduction-to-coding-in-r.html#cb17-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-10"><a href="introduction-to-coding-in-r.html#cb17-10" aria-hidden="true" tabindex="-1"></a><span class="co"># Save multiple R objects (note that this file is not provided, just example code is for future reference)</span></span>
<span id="cb17-11"><a href="introduction-to-coding-in-r.html#cb17-11" aria-hidden="true" tabindex="-1"></a><span class="fu">save</span>(object1, object2, <span class="st">&quot;multiple_objects.RData&quot;</span>)</span>
<span id="cb17-12"><a href="introduction-to-coding-in-r.html#cb17-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-13"><a href="introduction-to-coding-in-r.html#cb17-13" aria-hidden="true" tabindex="-1"></a><span class="co"># Save entire workspace</span></span>
<span id="cb17-14"><a href="introduction-to-coding-in-r.html#cb17-14" aria-hidden="true" tabindex="-1"></a><span class="fu">save.image</span>(<span class="st">&quot;entire_workspace.RData&quot;</span>)</span>
<span id="cb17-15"><a href="introduction-to-coding-in-r.html#cb17-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-16"><a href="introduction-to-coding-in-r.html#cb17-16" aria-hidden="true" tabindex="-1"></a><span class="co"># Load entire workspace</span></span>
<span id="cb17-17"><a href="introduction-to-coding-in-r.html#cb17-17" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="st">&quot;entire_workspace.RData&quot;</span>)</span></code></pre></div>
</div>
</div>
<div id="concluding-remarks-1" class="section level2 hasAnchor">
<h2>Concluding Remarks<a href="introduction-to-coding-in-r.html#concluding-remarks-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Together, this training module provides introductory level information on installing and loading packages in R. Scripting basics are also included, such as setting a working directory, importing and exporting files, and viewing data within the R console / RStudio environment. Additional resources that provide introductory-level information on coding in R include the following:</p>
<ul>
<li>Coursera provides a lot of materials on learning how to program in R: <a href="https://www.coursera.org/learn/r-programming" class="uri">https://www.coursera.org/learn/r-programming</a> &amp; <a href="https://www.coursera.org/courses?query=r" class="uri">https://www.coursera.org/courses?query=r</a></li>
<li>Stack overflow is a discussion forum for an online community of coders to discuss coding problems/challenges and ways to overcome these problems/challenges: <a href="https://stackoverflow.com/questions/1744861/how-to-learn-r-as-a-programming-language" class="uri">https://stackoverflow.com/questions/1744861/how-to-learn-r-as-a-programming-language</a></li>
<li>Wonderful tutorials are available online, like this one on ‘R for Data Science’: <a href="https://r4ds.had.co.nz/" class="uri">https://r4ds.had.co.nz/</a></li>
<li>BioConductor provides package-specific help: <a href="https://www.bioconductor.org/" class="uri">https://www.bioconductor.org/</a></li>
<li>An abundance of other resources are available online just by googling!</li>
</ul>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="index.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="data-organization-basics.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["TAME_Toolkit.pdf", "TAME_Toolkit.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "section"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
